include/zephyr/bluetooth/hci_types.h

```diff
@@ -2905,6 +2935,28 @@ struct bt_hci_evt_le_req_peer_sca_complete {
 	uint8_t  sca;
 } __packed;
 
+#define BT_HCI_EVT_LE_TRANSMIT_POWER_REPORT     0x21
+
+enum bt_hci_le_tx_power_report_reason {
+	/** Local Transmit power changed. */
+	BT_HCI_LE_TX_POWER_REPORT_REASON_LOCAL_CHANGED = 0,
+
+	/** Remote Transmit power changed. */
+	BT_HCI_LE_TX_POWER_REPORT_REASON_REMOTE_CHANGED = 1,
+
+	/** HCI_LE_Read_Remote_Transmit_Power_Level command completed. */
+	BT_HCI_LE_TX_POWER_REPORT_REASON_READ_REMOTE_COMPLETED = 2,
+};
+struct bt_hci_evt_le_transmit_power_report {
+	uint8_t	 status;
+	uint16_t handle;
+	enum bt_hci_le_tx_power_report_reason  reason;
```

Comment:
I don't think this is safe, and may not always work. Most platforms will use the native word width (e.g. 32 bits) for enum types. The only way to force the compiler to shrink the type to the minimum possible width is with the `__packed` annotation. Since these HCI structs themselves have a `__packed` annotation it *might* get "inherited" to the individual members, but I suspect it might also just be used to strip off unnecessary padding. Either way, this is not ideal: for these protocol structs we really do want to be explicit about the width of each member, so that the exactly match what's defined in the HCI specification, i.e. `uint8_t`, `uint16_t`, etc are really the appropriate types IMO.

Reply:
Hm, yeah, I see. But then what do you think I should do about the enum? have it be a bt_conn enum, with BT_HCI types, that is defined in hci_types.h but only used in conn.h?

Reply:
I agree that enums are nice when they can be used, however in this case, if you want to be consistent between the HCI struct and the API, both would need to use `uint8_t`, and in that case you're back to my proposal of using `#define` :)

Reply:
Alright :) In that case there will be no usage of these defines implemented though, don't see the benefit to have them defined then. Instead I'd just reference the spec and perhaps expand the explanation in the comment before the parameter. 
Let me know if I am missing something in that thought process.

Reply:
> Alright :) In that case there will be no usage of these defines implemented though, don't see the benefit to have them defined then. Instead I'd just reference the spec and perhaps expand the explanation in the comment before the parameter. Let me know if I am missing something in that thought process.

That's fine, though I do think they might be helpful to applications, so they don't need to hard-code values or create their own defines. In fact, this whole new feature submission should probably have been accompanied with a sample app that shows the intended usage of the new API. I.e. it's not just that there's no usage of the defines as the PR now stands, there's no usage of any part of the API, which is not really good when you consider of CI coverage and sample application code.

Reply:
I agree with that, but the feature can only be utilized minimally with the le commands, the real benefit of it comes with the nordic-only vendor-specific commands.

Reply:
> I agree with that, but the feature can only be utilized minimally with the le commands, the real benefit of it comes with the nordic-only vendor-specific commands.

If there's no value to non-Nordic platforms, then the feature doesn't belong upstream. If there is some value, sure, but you need to demonstrate that the code actually works. Is building the new code currently covered by any upstream test cases? In other words, when I see the CI checks passing has it actually tried to build the new code? We also have the bsim tests which could be used to implement runtime CI checks for the feature and have CI verify that the code doesn't just build but actually works.